CMakeLists.txt

```diff
@@ -30,160 +30,136 @@ set(IRODS_MIN_VERSION "4.2.0")
-                    "$ENV{IRODS_PATH}/lib/irods/externals/libboost_filesystem.a"
-                    "$ENV{IRODS_PATH}/lib/irods/externals/libboost_program_options.a"
-                    "$ENV{IRODS_PATH}/lib/irods/externals/libboost_regex.a"
-                    "$ENV{IRODS_PATH}/lib/irods/externals/libboost_system.a"
-                    "$ENV{IRODS_PATH}/lib/irods/externals/libboost_thread.a")
-        endif()
-    else()
-        set(irods_include_path_list "$ENV{IRODS_PATH}/lib/api/include"
-                "$ENV{IRODS_PATH}/lib/core/include"
-                "$ENV{IRODS_PATH}/server/icat/include"
-                "$ENV{IRODS_PATH}/lib/md5/include/"
-                "$ENV{IRODS_PATH}/lib/sha1/include/"
-                "$ENV{IRODS_PATH}/server/core/include/"
-                "$ENV{IRODS_PATH}/server/drivers/include/"
-                "$ENV{IRODS_PATH}/server/re/include/")
-        set(irods_link_obj_path "$ENV{IRODS_PATH}/lib/core/obj/libRodsAPIs.a")
-    endif()
-else()
-    message( FATAL_ERROR "IRODS_PATH is NOT defined, CMake will exit." )
+add_definitions(-DIRODS_HEADER_HPP)
+find_package(IRODS REQUIRED CONFIG)
+
+set(CMAKE_INSTALL_RPATH ${IRODS_EXTERNALS_FULLPATH_CLANG_RUNTIME}/lib)
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++")
+set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -stdlib=libc++")
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -stdlib=libc++")
+add_compile_options(-nostdinc++ -O0 -Wno-deprecated-register)
+
+set(CMAKE_C_COMPILER ${IRODS_EXTERNALS_FULLPATH_CLANG}/bin/clang)
+set(CMAKE_CXX_COMPILER ${IRODS_EXTERNALS_FULLPATH_CLANG}/bin/clang++)
+link_libraries(c++abi)
+
+remove_definitions(-DIRODS_HEADER_HPP)
+add_definitions(-DIRODS_42)
+set(irods_include_path_list
+    "${IRODS_INCLUDE_DIRS}"
+    "${IRODS_EXTERNALS_FULLPATH_CLANG}/include/c++/v1"
+    "${IRODS_EXTERNALS_FULLPATH_BOOST}/include"
+    "${IRODS_EXTERNALS_FULLPATH_JSON}/include"
+    )
+
+set(irods_link_obj_path
+    PRIVATE
+    irods_client
+    irods_common
+    irods_plugin_dependencies
+    "-lrt"
+    "-lcurl"
+    "${IRODS_EXTERNALS_FULLPATH_AVRO}/lib/libavrocpp.so"
+    "${IRODS_EXTERNALS_PACKAGE_ROOT}/jansson2.7-0/lib/libjansson.so"
+    "${IRODS_EXTERNALS_FULLPATH_BOOST}/lib/libboost_system.so"
+    "${IRODS_EXTERNALS_FULLPATH_BOOST}/lib/libboost_regex.so"
+    "${IRODS_EXTERNALS_FULLPATH_BOOST}/lib/libboost_thread.so"
+    "${IRODS_EXTERNALS_FULLPATH_BOOST}/lib/libboost_filesystem.so"
+    "/usr/lib/irods/plugins/network/libtcp_client.so"  # TODO
+    "/usr/lib/irods/plugins/auth/libnative_client.so"  # TODO
+    )
+
+if (NOT CPACK_PACKAGING_INSTALL_PREFIX)
+    set(CPACK_PACKAGING_INSTALL_PREFIX "/" CACHE STRING "Package root path. \"/\" is correct for normal package builds.." FORCE)
+    message(STATUS "Setting unspecified CPACK_PACKAGING_INSTALL_PREFIX to '${CPACK_PACKAGING_INSTALL_PREFIX}'. This is the correct setting for normal builds.")
+endif()
```

Comment:
Instead of doing this, you can probably `include(GNUInstallDirs)`. I usually like to do this shortly after the `project` call, or after including `${IRODS_TARGETS_PATH}`